arbor/memory/device_coordinator.hpp

```diff
@@ -81,7 +81,14 @@ class device_reference {
     }
 
     device_reference& operator=(const device_reference& ref) {
-        gpu_memcpy_d2d(pointer_, ref.pointer_, sizeof(T));
+        if (this != &ref) {
+            gpu_memcpy_d2d(pointer_, ref.pointer_, sizeof(T));
+        }
+        return *this;
+    }
+        
+    device_reference(const device_reference& ref) {
+        *this = ref;
     }
```

Comment:
Is this not incorrect? At this point `this->pointer_` is default initialized to an indeterminate value, and then the assignment will perform a GPU memory copy to outer space.

I think the static analyzer might be mollified with `device_reference(const device_reference&) = default`.